ansible/roles/gpu/tasks/nvidia-gpu.yaml

```diff
@@ -2,18 +2,111 @@
 - name: "NVIDIA Gpu - CentOS 7"
   include_tasks: nvidia-gpu-CentOS7.yaml
   when: ansible_distribution == "CentOS" and ansible_distribution_major_version == '7'
-- name: "NVIDIA Gpu - RHEL 7"
-  include_tasks: nvidia-gpu-RHEL7.yaml
-  when: ansible_distribution == 'RedHat' and ansible_distribution_major_version == '7'
-- name: "NVIDIA Gpu - RHEL 8"
-  include_tasks: nvidia-gpu-RHEL8.yaml
-  when: ansible_distribution == 'RedHat' and ansible_distribution_major_version == '8'
+
+- name: "NVIDIA Gpu - RHEL"
+  include_tasks: nvidia-gpu-RHEL.yaml
+  when: ansible_distribution == 'RedHat'
+
 - name: "NVIDIA Gpu - SLES 15"
   include_tasks: nvidia-gpu-SLES15.yaml
   when: ansible_distribution == "SLES" and ansible_distribution_major_version == '15'
+
 - name: "NVIDIA Gpu - Flatcar"
   include_tasks: nvidia-gpu-Flatcar.yaml
   when: ansible_distribution == "Flatcar"
+
 - name: "NVIDIA Gpu - Ubuntu 20"
   include_tasks: nvidia-gpu-Ubuntu20.yaml
   when: ansible_distribution == "Ubuntu"
+
+# disable with these instructions https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#runfile-nouveau
+- name: create nouveau blacklist file
+  copy:
+    src: ./blacklist-nouveau.conf
+    dest: /etc/modprobe.d/blacklist-nouveau.conf
+
+- name: regenerate the kernel initramfs
+  command: dracut --force
+  become: yes
+  when:
+    - ansible_os_family == "RedHat"
+
+- name: regenerate the kernel initramfs
+  command: update-initramfs -u
+  become: yes
+  when: ansible_os_family == "Debian"
+
+# to pick up the changes to unload nouveau
+# some hardware like aws g4dn.2xlarge require this
+- name: unconditionally reboot the machine with all defaults
+  ansible.builtin.reboot:
+  when: sysprep
+
+# Download and install gpu runfile
+- name: create nvidia install directory
+  file:
+    path: /opt/dkp/nvidia
+    state: directory
```

Comment:
ansible re-runs gives error when `/opt/dkp/nvidia` already exists. Delete `/opt/dkp/nvidia` to delete old installations first or add file exists check.